### PR TITLE
add reward check in minicluster test

### PR DESCRIPTION
**Pull request by huof6829**
# Description

Fixes  https://github.com/iotexproject/iotex-core/issues/2901

## Type of change
Please delete options that are not relevant.
- [] Bug fix (non-breaking change which fixes an issue)
- [] New feature (non-breaking change which adds functionality)
- [] Code refactor or improvement
- [] Breaking change (fix or feature that would cause a new or changed behavior of existing functionality)
- [] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
- [] make test
- [] fullsync
- [] Other test (please specify)

**Test Configuration**:
- Firmware version:
- Hardware:
- Toolchain:
- SDK:

# Checklist:
- [] My code follows the style guidelines of this project
- [] I have performed a self-review of my code
- [] I have commented my code, particularly in hard-to-understand areas
- [] I have made corresponding changes to the documentation
- [] My changes generate no new warnings
- [] I have added tests that prove my fix is effective or that my feature works
- [] New and existing unit tests pass locally with my changes
- [] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by raullenchai — APPROVED** [review]

File: tools/minicluster/minicluster.go [hunk 1]
```diff
@@ -390,6 +393,42 @@ func main() {
 
 			log.S().Info("Fp token transfer test pass!")
 		}
+
+		registries := make([]*protocol.Registry, _numNodes)
+		for i := 0; i < _numNodes; i++ {
+			registries[i] = svrs[i].ChainService(configs[i].Chain.ID).Registry()
```
Comment (CoderZhi) [inline tools/minicluster/minicluster.go:399]: will a local parameter sufficient?

File: tools/minicluster/minicluster.go [hunk 2]
```diff
@@ -390,6 +393,42 @@ func main() {
...
+			})
+			ctx = genesis.WithGenesisContext(
+				protocol.WithRegistry(ctx, registries[i]),
+				chains[i].Genesis(),
+			)
+			ctx = protocol.WithFeatureCtx(protocol.WithFeatureWithHeightCtx(ctx))
+
+			rp := rewarding.FindProtocol(registries[i])
+			if rp == nil {
+				log.S().Error("rolldpos is not registered.")
```
Comment (CoderZhi) [inline tools/minicluster/minicluster.go:412]: if `rp` is nil, how could line 415 works?

**Review by CoderZhi — CHANGES_REQUESTED** [review]

File: tools/minicluster/minicluster.go [hunk 3]
```diff
@@ -390,6 +393,42 @@ func main() {
...
+			ctx = protocol.WithFeatureCtx(protocol.WithFeatureWithHeightCtx(ctx))
+
+			rp := rewarding.FindProtocol(registries[i])
+			if rp == nil {
+				log.S().Fatal("rolldpos is not registered.")
+			}
+
+			blockReward, err := rp.BlockReward(ctx, sfs[i])
+			if err != nil {
+				log.S().Error("Failed to get block reward.", zap.Error(err))
```
Comment (CoderZhi) [inline tools/minicluster/minicluster.go:417]: same problem here

File: tools/minicluster/minicluster.go [hunk 4]
```diff
@@ -390,6 +393,42 @@ func main() {
...
+			}
+			if blockReward == configs[i].Genesis.BlockReward() {
+				log.S().Error("actual block reward is incorrect.")
```
Comment (CoderZhi) [inline tools/minicluster/minicluster.go:420]: and here

File: tools/minicluster/minicluster.go [hunk 5]
```diff
@@ -390,6 +393,42 @@ func main() {
...
+			}
+
+			epochReward, err := rp.EpochReward(ctx, sfs[i])
+			if err != nil {
+				log.S().Error("Failed to get epoch reward.", zap.Error(err))
```
Comment (CoderZhi) [inline tools/minicluster/minicluster.go:425]: and here

**Review by CoderZhi — APPROVED** [review]

**Pull request merged**